Purchases/Purchasing/PurchasesOrchestrator.swift

```diff
@@ -194,9 +194,9 @@ class PurchasesOrchestrator {
     func purchase(package: Package, completion: @escaping PurchaseCompletedBlock) {
         // todo: clean up, move to new class along with the private funcs below
         if #available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *),
```

Comment:
should we be checking the `useStoreKit2` variable here?

Reply:
That's a very good question.
This is called from the `public`  `Purchases.purchase(package:completion:)`. I wonder if there's a situation where users can have a `Package` with an SK1 `StoreProduct`, but with `useStoreKit2` set to `true`. In that case this would fail.

I'm guessing @aboedo did this on purpose so that no matter what the setting is, we purchase based on what's inside the `Package`.

Reply:
> I'm guessing @aboedo did this on purpose so that no matter what the setting is, we purchase based on what's inside the `Package`.

Yeah, that was my thinking. At the time I wasn't sure whether we'd need a `useStoreKit2` flag in the first place, and I don't think we'll need it forever anyway. Hopefully we should be able to kill it or make it internal once we're more familiar with how reliable SK2 is in production. 

As for whether it'd be possible to run into this situation, well, the `init` from `Package` is internal, so it should only be created by the SDK. And the SDK would only create it based on the current `useStoreKit2` value. 

So, barring the possibility of modifying the SDK (it's open source after all) or using KVC (these are NSObjects), you shouldn't run into a troublesome situation.